text/0000-license-rfcs.md

```diff
@@ -0,0 +1,162 @@
+<LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+option. This file may not be copied, modified, or distributed
+except according to those terms.
+-->
+
+- Feature Name: license_rfcs
+- Start Date: 2017-06-26
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Introduce a move to dual-MIT/Apache2 licensing terms to the Rust RFCs repo, by
+requiring them for all new contributions, and asking previous contributors to
+agree on the new license.
+
+# Disclaimer
+[disclaimer]: #disclaimer
+
+This RFC is not authored by a lawyer, so its reasoning may be wrong.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the Rust RFCs repo is in a state where no clear open source license
+is specified.
+
+The current legal base of the the RFCs repo is the "License Grant to Other
+Users" from the [Github ToS]`*`:
+
+```
+Any Content you post publicly, including issues, comments, and contributions to other Users' repositories, may be viewed by others. By setting your repositories to be viewed publicly, you agree to allow others to view and "fork" your repositories (this means that others may make their own copies of your Content in repositories they control).
+
+If you set your pages and repositories to be viewed publicly, you grant each User of GitHub a nonexclusive, worldwide license to access your Content through the GitHub Service, and to use, display and perform your Content, and to reproduce your Content solely on GitHub as permitted through GitHub's functionality.
+```
+
+These terms may be sufficient for display of the rfcs repository on Github, but
+it limits contributions and use, and even poses a risk.
+
+The Github ToS grant only applies towards reproductions through the Github
+Service. Hypothetically, if the Github Service ceases at some point in the
+future, without a legal successor offering a replacement service, the RFCs may
+not be redistributed any more.
+
+Second, there are companies which have set up policies that limit their
+employees to contribute to the RFCs repo in this current state.
+
+Third, there is the possibility that Rust may undergo standardisation and
+produce a normative document describing the language.
+Possibly, the authors of such a document may want to include text from RFCs.
+
+Fourth, the spirit of the Rust project is to be open source, and the current
+terms don't fulfill any popular open source definition.
+
+`*`: The Github ToS is licensed under the [Creative Commons Attribution license](https://creativecommons.org/licenses/by/4.0/)
+
+[Github ToS]: https://help.github.com/articles/github-terms-of-service/#5-license-grant-to-other-users
+
+# Detailed design
+[design]: #detailed-design
+
+After this RFC has been merged, all new RFCs will be required to be
+dual-licensed under the MIT/Apache2. This includes RFCs currently being
+[considered for merging].
+
+This puts a responsibility on anyone who is merging RFCs to check for a
+license header.
```

Comment:
rust-lang/rust has a tool for this, but the rfc process is highly manual, and the way PRs get merged is different as well. Introducing a tool only for this purpose seems a bit much imo. Maybe we can do this after the transition period is over, and we still see files without a header being proposed.